grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -13,6 +14,28 @@ class ProductService {
 	PermissionService permissionService
 	SubscriptionService subscriptionService
 
+	void removeUsersProducts(String username) {
+		def all = Product.findAllByOwner(username)
+		all.toArray().each{ Product product ->
+			product.delete(flush: true)
```

Comment:
I think this needs to be done last because `flush: true` will attempt to perform the SQL deletion immediately and will fail because there are foreign keys still linked to it.